qiskit/_quantumprogram.py

```diff
@@ -464,31 +470,50 @@ def load_qasm_text(self, qasm_string, name=None,
     # methods to get elements from a QuantumProgram
     ###############################################################
 
-    def get_quantum_register(self, name):
+    def get_quantum_register(self, name=None):
         """Return a Quantum Register by name.
 
         Args:
-            name (str): the name of the quantum register
+            name (hashable or None): the name of the quantum register. If None and there is only
+                one quantum register available, returns that one.
         Returns:
-            QuantumRegister: The quantum register with this name
+            QuantumRegister: The quantum register with this name.
         Raises:
             KeyError: if the quantum register is not on the quantum program.
+            QISKitError: if the register does not exist in the program.
         """
+        if name is None:
+            qrms = list(self.get_quantum_register_names())
+            if len(qrms) == 1:
+                name = qrms[0]
+            else:
+                raise QISKitError("You have to select a quantum register to get when there is more"
+                                  "than one available")
```

Comment:
Nit: perhaps we can generalize this pattern in a small function `name = get_name_if_there_is_only_one_item_or_raise_an_error(some_list)` ? It seems to be used in a number of places - not sure if they are enough, though.

Reply:
I went for `get_single_name`, it seems shorter :) (see a011417)